modules/scatteredBidAdapter.js

```diff
@@ -0,0 +1,175 @@
+// jshint esversion: 6, es3: false, node: true
+'use strict';
+
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { config } from '../src/config.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { mergeDeep, deepAccess, parseSizesInput, parseUrl, deepSetValue, logInfo, createTrackPixelHtml } from '../src/utils.js';
+
+const { getConfig } = config;
+
+const BIDDER_CODE = 'scattered';
+
+export function buildImps(validBidRequests) {
+  return validBidRequests.map((bid, index) => {
+    let imp = {
+      id: `${index + 1}`
+    };
+    const bannerParams = deepAccess(bid, 'mediaTypes.banner');
+    if (bannerParams && bannerParams.sizes) {
+      const sizes = parseSizesInput(bannerParams.sizes);
+      const format = sizes.map(size => {
+        const [width, height] = size.split('x');
+        const w = parseInt(width, 10);
+        const h = parseInt(height, 10);
+        return { w, h };
+      });
+
+      imp.banner = {
+        format
+      };
+    };
+    return imp;
+  });
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  // 1.
+  isBidRequestValid: function (bid) {
+    const bidderDomain = deepAccess(bid, 'params.bidderDomain')
+    if (bidderDomain === undefined || bidderDomain === '') {
+      return false
+    }
+
+    const sizes = deepAccess(bid, 'mediaTypes.banner.sizes')
+    if (sizes === undefined || sizes.length < 1) {
+      return false
+    }
+
+    return true
+  },
+
+  // 2.
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const ortb2Obj = bidderRequest.ortb2 || {};
+    let { user } = ortb2Obj
+
+    let site = getConfig('site') || {};
+    if (ortb2Obj.site) {
+      mergeDeep(site, ortb2Obj.site)
+    }
+    // site.page    URL of the page where the impression will be shown.
+    // site.domain  Domain of the site (e.g., “mysite.foo.com”).
+    if (!site.page && bidderRequest.refererInfo) {
+      site.page = bidderRequest.refererInfo.referer;
+      site.domain = parseUrl(bidderRequest.refererInfo.referer).hostname
+    }
+
+    const bidderDomain = getKeyOnAny(validBidRequests, 'params.bidderDomain')
+
+    const device = getConfig('device') || {};
+    device.w = device.w || window.innerWidth;
+    device.h = device.h || window.innerHeight;
+    device.ua = device.ua || navigator.userAgent;
+
+    const imps = buildImps(validBidRequests)
+
+    const tid = getKeyOnAny(validBidRequests, 'transactionId');
```

Comment:
source.tid should be the auctionId, and kept separate from imp transaction ids - see [this issue](https://github.com/prebid/Prebid.js/issues/8573)